src/lib/core/mp_interval.c

```diff
@@ -99,13 +99,28 @@ interval_pack(char *data, const struct interval *itv)
 }
 
 struct interval *
-interval_unpack(const char **data, struct interval *itv)
+interval_unpack(const char **data, uint32_t len, struct interval *itv)
 {
+	const char *end = *data + len;
+	uint32_t count = mp_load_u8(data);
```

Comment:
AFAIU you should check that `len` is > 0 before loading `data`.

Reply:
Ditto.

Reply:
Added an assertion.